src/rrddump.ml

```diff
@@ -40,11 +40,12 @@ let text_export rrd =
 
 let with_input_file filename f =
   let ic = open_in filename in
-finally (fun () -> f ic) (fun () -> close_in ic)
```

Comment:
This is not the same. When `f` succeeds, the channel is no longer closed. You can use finally from  library or define it locally. Below is the code from the upcoming OCaml release

```
let finally work ~(always: unit -> unit) =
  match work () with
  | result -> always (); result
  | exception work_exn ->
    always ();
    raise work_exn
```